lib/Git/Wrapper.pm

```diff
@@ -87,25 +91,41 @@ sub RUN {
 
     my ($wtr, $rdr, $err);
 
+    my $file;
     local *TEMP;
-    if ($^O eq 'MSWin32' && defined $stdin) {
-      my $file = File::Temp->new;
-      $file->autoflush(1);
-      $file->print($stdin);
-      $file->seek(0,0);
-      open TEMP, '<&=', $file;
-      $wtr = '<&TEMP';
-      undef $stdin;
+    if (defined $stdin) {
+      if ($^O eq 'MSWin32') {
+        $file = File::Temp->new;
+        $file->autoflush(1);
+        binmode $file;
+        print { $file } $stdin;
+        seek $file, 0, 0;
+        open TEMP, '<&=', $file;
+        $wtr = '<&TEMP';
+        undef $stdin;
+      }
     }
 
     $err = Symbol::gensym;
 
-    print STDERR join(' ',@cmd),"\n" if $DEBUG;
-
     # Prevent commands from running interactively
     local $ENV{GIT_EDITOR} = ' ';
 
+    # best attempt at encoding
+    @cmd = map { Encode::encode('UTF-8', $_) } @cmd
+      if USE_ENCODE;
+
+    print STDERR join(' ',@cmd),"\n" if $DEBUG;
+
+    system 'chcp', '65001';
```

Comment:
What's `chcp`? That doesn't seem to be available on any system I have easy access to.